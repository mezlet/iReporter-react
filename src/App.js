import React from "react";

import Route from "./routes/Routes";

const App = () => (
  <div>
    <Route />
  </div>
);

export default App;
