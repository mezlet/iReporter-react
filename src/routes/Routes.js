/* eslint-disable import/no-named-as-default */
import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import HomePage from '../components/containers/HomePage/Home';
import connectSignup from '../components/containers/SignUp/SignUp';
import connectLogin from '../components/containers/Login/Login';
import CreateReport from '../components/containers/CreateReport/CreateReport';
import Profile from '../components/containers/Profile/Profile';
import NewReportView from '../components/containers/NewReportView/NewReportView';
import ViewIncident from '../components/containers/ViewIncident/ViewIncident';
import ViewRedFlag from '../components/containers/ViewRedFlag/ViewRedFlag';
import ViewIntervention from '../components/containers/ViewIntervention/ViewIntervention';
import EditReport from '../components/containers/EditReport/EditReport';
import UserIncident from '../components/containers/UserIncident/UserIncident';
import AuthenticateRoute from '../hoc/AuthenticateRoute';
import Logout from '../components/containers/Logout/Logout';
import AdminPage from '../components/containers/Admin/AdminPage';

const history = createBrowserHistory();

const route = () => (
  <BrowserRouter>
    <Route history={history}>
      <Switch>
        <Route path="/" component={HomePage} exact />
        <Route path="/signup" component={connectSignup} exact />
        <Route path="/login" component={connectLogin} exact />
        <Route path="/logout" component={Logout} exact />
        <AuthenticateRoute path="/create" component={CreateReport} exact />
        <AuthenticateRoute path="/profile" component={Profile} exact />
        <AuthenticateRoute path="/admin" component={AdminPage} exact />
        <AuthenticateRoute
          path="/edit-incident/:id"
          component={EditReport}
          exact
        />
        <AuthenticateRoute
          path="/new-report-view/:id"
          component={NewReportView}
          exact
        />
        <AuthenticateRoute path="/view-all" component={ViewIncident} exact />
        <AuthenticateRoute
          path="/view-red-flags"
          component={ViewRedFlag}
          exact
        />
        <AuthenticateRoute
          path="/view-interventions"
          component={ViewIntervention}
          exact
        />
        <AuthenticateRoute path="/record" component={UserIncident} exact />
      </Switch>
    </Route>
  </BrowserRouter>
);

export default route;
