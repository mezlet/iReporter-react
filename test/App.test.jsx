import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from '../src/App';
import Routes from '../src/routes/Routes';

const wrapper = mount(<App />);
describe('<App/> rendering', () => {
  it('should render one <Routes> component', () => {
    expect(wrapper.find(Routes)).toHaveLength(1);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<App />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
