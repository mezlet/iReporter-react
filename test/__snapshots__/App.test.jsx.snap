// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> rendering matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <route />
  </BrowserRouter>
  <ToastContainer
    autoClose={2000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</Provider>
`;
